assignment/assignment/LoginViewController.swift

```diff
@@ -0,0 +1,307 @@
+            .underlineStyle: NSUnderlineStyle.single.rawValue
+        ]
+        
+        let attributedTitle = NSMutableAttributedString(string: title, attributes: attributes)
+        $0.setAttributedTitle(attributedTitle, for: .normal)
+    }
+    
+    
+    let spaceView = UIView().then {
+        $0.backgroundColor = UIColor(named: "gray2")
+    }
+    
+    
+    let eyeButton = UIButton(type: .custom)
+    let xCircleButton = UIButton(type: .custom)
+    
+    
+    //MARK: - ViewDidLoad
+    override func viewDidLoad() {
+        super.viewDidLoad()
+        
+        view.backgroundColor = .black
+        idTextFieldView.delegate = self
+        passwordTextFieldView.delegate = self
+        
+        addSubViews()
+        setConstraints()
+        
+        // 패스워드 텍스트 필드 설정
+        eyeButton.setImage(UIImage(named: "eyeIcon"), for: .normal)
+        eyeButton.addTarget(self, action: #selector(togglePasswordView), for: .touchUpInside)
+        eyeButton.snp.makeConstraints { make in
+            make.trailing.equalTo(passwordTextFieldView.snp.trailing).offset(-20)
+            make.centerY.equalTo(passwordTextFieldView)
+            make.width.height.equalTo(24)
+        }
+        
+        xCircleButton.setImage(UIImage(named: "xCircle"), for: .normal)
+        xCircleButton.snp.makeConstraints { make in
+            make.trailing.equalTo(eyeButton.snp.leading).offset(-20)
+            make.centerY.equalTo(passwordTextFieldView)
+            make.width.height.equalTo(24)
+        }
+
+        idTextFieldView.addTarget(self, action: #selector(textFieldDidChange), for: .editingChanged)
+        passwordTextFieldView.addTarget(self, action: #selector(textFieldDidChange), for: .editingChanged)
+        xCircleButton.addTarget(self, action: #selector(handleXCircleButtonTap), for: .touchUpInside)
+        
+        loginButton.addTarget(self, action: #selector(handleLogin), for: .touchUpInside)
+        makeAccount.addTarget(self, action: #selector(presentModalView), for: .touchUpInside)
+
+        
+    
+    }
+    
+    //MARK: - AddSubview
+    func addSubViews() {
+        let views = [
+            loginLabel,
+            idTextFieldView,
+            passwordTextFieldView,
+            loginButton,
+            findId,
+            findPw,
+            noAccount,
+            makeAccount,
+            spaceView,
+            eyeButton,
+            xCircleButton
+        ]
+        views.forEach {
+            view.addSubview($0)
+        }
+    }
+    
+    
+    //MARK: - layout
+    func setConstraints() {
+        loginLabel.snp.makeConstraints { make in
+            make.top.equalTo(view.safeAreaLayoutGuide).offset(90)
+            make.centerX.equalTo(view)
+        }
+        
+        idTextFieldView.snp.makeConstraints { make in
+            make.top.equalTo(loginLabel.snp.bottom).offset(20)
+            make.centerX.equalTo(view)
+            make.width.equalTo(335)
```

Comment:
휴대폰 화면이 바뀌어도 해당 컴포넌트들의 크기가 유동적으로 바뀔수 있도록 
```suggestion
            make.leading.trailing.equalToSuperView().inset(20)
            //20은 원하는 숫자로 변경
```
이런식으로 코드를 쓰는것은 어떨까요?? 휴대폰 화면이 작아질경우 컴포넌트가 화면 밖으로 넘어가는 것을 막을 수 있을 것 같습니다!